app/controllers/users_controller.rb

```diff
@@ -0,0 +1,31 @@
+class UsersController < ApplicationController
+    def new 
+    end
+    def create
+      @user = User.new(whitelisted_post_params)
+    #   @user=User.new(username: params[:username], email: params[:email], password: params[:password])  
+      if @user.save
+        redirect_to new_user_path
+      else
+        render :new
+      end
+    end
+
+    def edit
+        @user=User.find(params[:id])
+    end
+
+    def update
+        @user=User.find(params[:id])
+        @user.update(whitelisted_post_params)
+
```

Comment:
for `update` action:
- [ ] `redirect_to edit_user_path` if user was saved
- [ ] `render :edit` if user was not saved